uspace/lib/posix/meson.build

```diff
@@ -29,6 +29,7 @@
 
 includes += include_directories('include/posix', 'include')
 c_args += [ '-fno-builtin', '-D_XOPEN_SOURCE' ]
+allow_shared = true
```

Comment:
Did you test this with coastline?

Reply:
'coastline'?

but I did test it with Rust programs, and also test-libposix passes with dynamic linking enabled

Reply:
"Coastline" is the home of ports that libposix was made for. Separate repository in the HelenOS project. I'll give it a try tomorrow, to make sure there's no weirdness.

Reply:
To be fair, I think all libraries should be shared anyway. Dunno if there are any lingering issues or if it's just the matter of prudence.

Reply:
Yeah, I want to look into this deeper in a separate PR. I just wanted libposix for my Rust programs, but I guess I could also just stay with static link until I figure out the uspace/lib/meson.build rewrite